ui/src/hooks.js

```diff
@@ -190,7 +190,18 @@ export const useChartSeries = ({
   );
 
   const isLoading = queries.find((query) => query.isLoading);
-  const queriesData = queries.map((query) => query.data);
+  const queriesData = queries
+    .map((query) => {
+      return query.data;
+    })
+    /* useQueries is running the requests in paralel and given that
+     * in transformPrometheusDataToSeries (which is a generic function used by multiple charts)
+     * we make an assumption on the order of responses
+     * then we need to make sure that the average query is the second one in the array
+     * That is achieved by giving a key param to the response object (e.g. 'cpuUsage' and 'cpuUsageAvg')
+     * and sorting the array alphanumerically on its 'key' property
+     */
+    .sort((query1, query2) => (query1.key > query2.key ? 1 : -1));
```

Comment:
Yes, it would've been great given that `useQueries` returned a keyed object (based on queryKey or whatever) or if they labeled in any way which response corresponds to which request. It is possible that they intended to have the order of array elements to be the same as the ones of requests (because it's always the case on first load), but apparently on the second one they flip